### ISSUE TITLE
Digital Ocean Ubuntu 15.10 x64 permission issues

### ISSUE THREAD

**Issue by ericwooley** [blockstack/blockstore#167]
@muneeb-ali  I tried installing on a fresh droplet with Ubuntu 15.10 x64 (installation through pip), and got this stack trace

```
root@blockstore:~# cat ~/.blockstore/blockstore.log.indexer
Traceback (most recent call last):
  File "/usr/local/bin/blockstored", line 4, in <module>
    __import__('pkg_resources').run_script('blockstore==0.0.6', 'blockstored')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 735, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1652, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/EGG-INFO/scripts/blockstored", line 36, in <module>
    run_blockstored()
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/blockstore/blockstored.py", line 1669, in run_blockstored
    run_server( testset=testset )
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/blockstore/blockstored.py", line 1428, in run_server
    blockstored = subprocess.Popen( api_server_command, shell=False)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 13] Permission denied
Traceback (most recent call last):
  File "/usr/local/bin/blockstored", line 4, in <module>
    __import__('pkg_resources').run_script('blockstore==0.0.6', 'blockstored')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 735, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1652, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/EGG-INFO/scripts/blockstored", line 36, in <module>
    run_blockstored()
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/blockstore/blockstored.py", line 1669, in run_blockstored
    run_server( testset=testset )
  File "/usr/local/lib/python2.7/dist-packages/blockstore-0.0.6-py2.7.egg/blockstore/blockstored.py", line 1428, in run_server
    blockstored = subprocess.Popen( api_server_command, shell=False)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 13] Permission denied
```

**Comment by justinguy** [comment]
@ericwooley Did you run as root (or sudo)?

**Comment by muneeb-ali** [comment]
Thanks, I'll try it out. 

I'm also staring a separate issue for pre-build Debian image.

**Comment by ericwooley** [comment]
@justinguy Yea, the default login is root. 

But just to be sure I tried with sudo, and got the same.

**Comment by justinguy** [comment]
Interesting, I read the error as the OS denying permission to start the subprocess. From the Python documentation:

> On Unix, the new executable is loaded into the current process, and will have the same process id as the caller.

For what it is worth, I just did this on Linode about a week ago but using Debian Jessie and didn't have this issue.

**Comment by ericwooley** [comment]
I am trying it with ubuntu 14 right now, see if it happens. IIRC doesn't AWS uses a modified ubuntu image to start for security. Maybe it has to do with that?

**Comment by ericwooley** [comment]
When installing on 14, off of master, i get:
```
Installed distribution bitcoin 1.1.38 conflicts with requirement bitcoin==1.1.36
```

**Comment by justinguy** [comment]
Do they run SELinux? You can try `cat /selinux/enforce` and see if the output is 1.

**Comment by ericwooley** [comment]
`cat: /selinux/enforce: No such file or directory`

I also just got it running on ubuntu 14, through the pip install, not through the git clone.

**Comment by justinguy** [comment]
Nice, pre-built images are going to be a lifesaver.

**Comment by ericwooley** [comment]
@justinguy @muneeb-ali I got it to work in 15, not sure why it didn't work before, maybe because I did a pip install and then tried to install from master. I made a little install script for the future, in case anyone else runs into the issue.

```bash
sudo apt-get update
sudo apt-get -y install python-dev git python-pip
git clone https://github.com/blockstack/blockstore.git
cd blockstore
pip install -r requirements.txt
python setup.py build
sudo python setup.py install
cd ..
```


If you run into an error about testnet, from https://github.com/blockstack/blockstore/issues/163 you can run
`sed -i -e "s/testset = False/#testset = False/" ~/.blockstore/blockstore.ini`

**Comment by justinguy** [comment]
@ericwooley Awesome, thanks for sharing the script! Pull #165 fixes the testset issue in #163 so once that gets merged it should be better.
